specification/search/data-plane/Azure.Search/preview/2020-06-30-Preview/searchservice.json

```diff
@@ -8843,7 +8843,55 @@
         "code": {
           "type": "string",
           "readOnly": true,
-          "description": "One of a server-defined set of error codes."
+          "description": "One of a server-defined set of error codes.",
+          "enum": [
+            "OperationNotAllowed",
+            "FeatureNotSupportedInApiVersion",
+            "FeatureNotSupportedInService",
+            "InvalidRequestParameter",
+            "MissingRequiredParameter",
+            "InvalidName",
+            "InvalidPayloadFormat",
+            "LimitExceeded"
+          ],
+          "x-ms-enum": {
+            "name": "",
+            "modelAsExtensible": true,
+            "values": [
+              {
+                "value": "OperationNotAllowed",
+                "description": "Operation is not allowed."
+              },
+              {
+                "value": "FeatureNotSupportedInApiVersion",
+                "description": "Operation is not supported in the current API version."
+              },
+              {
+                "value": "FeatureNotSupportedInService",
+                "description": "Operation is not supported in the current service."
+              },
+              {
+                "value": "InvalidRequestParameter",
+                "description": "Request contains an invalid parameter."
+              },
+              {
+                "value": "MissingRequiredParameter",
+                "description": "Required parameter is missing from the request."
+              },
+              {
+                "value": "InvalidName",
+                "description": "Request contains an entity name that is syntactically wrong."
```

Comment:
Is the pattern consistent across resource types you could document what is valid? We don't do client-side validation by default to avoid tight-coupling (though we are discussing whether we should to some extent).